ec/ecdatasource/deploymentdatasource/schema_apm.go

```diff
@@ -18,72 +18,116 @@
+				Computed:    true,
 			},
 			"healthy": {
-				Type:     schema.TypeBool,
-				Computed: true,
+				Type:        types.BoolType,
+				Description: "Resource kind health status.",
+				Computed:    true,
 			},
 			"http_endpoint": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:        types.StringType,
+				Description: "HTTP endpoint for the resource kind.",
+				Computed:    true,
 			},
 			"https_endpoint": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:        types.StringType,
+				Description: "HTTPS endpoint for the resource kind.",
+				Computed:    true,
 			},
 			"ref_id": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:        types.StringType,
+				Description: "User specified ref_id for the resource kind.",
+				Computed:    true,
 			},
 			"resource_id": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:        types.StringType,
+				Description: "The resource unique identifier.",
+				Computed:    true,
 			},
 			"status": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:        types.StringType,
+				Description: "Resource kind status (for example, \"started\", \"stopped\", etc).",
+				Computed:    true,
 			},
 			"version": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:        types.StringType,
+				Description: "Elastic stack version.",
+				Computed:    true,
 			},
 			"topology": apmTopologySchema(),
-		},
+		}),
 	}
 }
 
-func apmTopologySchema() *schema.Schema {
-	return &schema.Schema{
-		Type:     schema.TypeList,
-		Computed: true,
-		Elem: &schema.Resource{
-			Schema: map[string]*schema.Schema{
-				"instance_configuration_id": {
-					Type:     schema.TypeString,
-					Computed: true,
-				},
-				"size": {
-					Type:     schema.TypeString,
-					Computed: true,
-				},
-				"size_resource": {
-					Type:     schema.TypeString,
-					Computed: true,
-				},
-				"zone_count": {
-					Type:     schema.TypeInt,
-					Computed: true,
-				},
+func apmResourceInfoAttrTypes() map[string]attr.Type {
+	return apmResourceInfoSchema().Attributes.Type().(types.ListType).ElemType.(types.ObjectType).AttrTypes
+}
+
+func apmTopologySchema() tfsdk.Attribute {
+	return tfsdk.Attribute{
+		Description: "Node topology element definition.",
+		Computed:    true,
+		Validators:  []tfsdk.AttributeValidator{listvalidator.SizeAtMost(1)},
+		Attributes: tfsdk.ListNestedAttributes(map[string]tfsdk.Attribute{
+			"instance_configuration_id": {
+				Type:        types.StringType,
+				Description: "Controls the allocation of this topology element as well as allowed sizes and node_types. It needs to match the ID of an existing instance configuration.",
+				Computed:    true,
+			},
+			"size": {
+				Type:        types.StringType,
+				Description: "Amount of resource per topology element in the \"g\" notation.",
```

Comment:
What's the "g" notation. 

```suggestion
				Description: "Amount of size_resource in Gigabytes. For example \"4g\".",
```